spi/src/main/java/org/eclipse/dataspaceconnector/spi/types/domain/contract/Contract.java

```diff
@@ -0,0 +1,218 @@
+/*
+ *  Copyright (c) 2021 Daimler TSS GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Daimler TSS GmbH - Initial API and Implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.spi.types.domain.contract;
+
+import org.eclipse.dataspaceconnector.policy.model.Policy;
+import org.eclipse.dataspaceconnector.spi.types.domain.asset.Asset;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * {@link Contract} to regulate data transfer between two parties.
+ */
+public class Contract {
+
+    private final String id;
+    private final String providerId;
+    private final String consumerId;
+    private final long contractSigningDate;
+    private final long contractStartDate;
+    private final long contractEndDate;
+    private final List<Asset> assets;
```

Comment:
We expect the `Contract` to serialize all the `Asset`s or just their ids?

Reply:
As the `Contract` represents is a legal document I'd put the `Asset` in there. In case there is a disagreement between two parties, the persistent `Contract` would contain all the available `Assets` data at the time the contract was signed/persisted.